addons/point_of_sale/static/src/app/services/data_service.js

```diff
@@ -364,6 +364,15 @@ export class PosData extends Reactive {
 
             this.models[model].addEventListener("update", (params) => {
                 const record = this.models[model].get(params.id).raw;
+
+                for (const [key, value] of Object.entries(record)) {
+                    if (value instanceof Base) {
+                        record[key] = value.id;
+                    } else if (Array.isArray(value) && value[0] instanceof Base) {
+                        record[key] = value.map((v) => v.id);
+                    }
+                }
+
```

Comment:
Instead of manually casting the relational fields to their ids, can't you just call `serialize` here?

Reply:
Isn't a record,  is data that coming from the server or update data, and sometime update data contains record itself and we just need its id